server/sonar-server/src/main/java/org/sonar/server/sticker/ws/MeasureAction.java

```diff
@@ -19,54 +19,195 @@
+      LiveMeasureDto measure = getMeasure(dbSession, project, metricKey);
+      write(generateSvg(metric, measure), response.stream().output(), UTF_8);
+    } catch (StickerException | ForbiddenException | NotFoundException e) {
+      write(svgGenerator.generateError(e.getMessage()), response.stream().output(), UTF_8);
+    }
+  }
+
+  private String generateSvg(MetricDto metric, LiveMeasureDto measure) {
+    String metricType = metric.getValueType();
+    String metricName = METRIC_NAME_BY_KEY.get(metric.getKey());
+    switch (ValueType.valueOf(metricType)) {
+      case INT:
+        return svgGenerator.generateBadge(metricName, formatNumber(getNonNullValue(metric, measure, LiveMeasureDto::getValue).longValue()), Color.DEFAULT);
+      case PERCENT:
+        return svgGenerator.generateBadge(metricName, PERCENT_FORMATTER.format(getNonNullValue(metric, measure, LiveMeasureDto::getValue)), Color.DEFAULT);
+      case LEVEL:
+        return generateQualityGate(metric, measure);
+      case WORK_DUR:
+        return svgGenerator.generateBadge(metricName, durations.format(Duration.create(getNonNullValue(metric, measure, LiveMeasureDto::getValue).longValue())),
+          Color.DEFAULT);
+      case RATING:
+        return generateRating(metric, measure);
+      default:
+        throw new IllegalStateException(format("Invalid metric type '%s'", metricType));
+    }
+  }
+
+  private String generateQualityGate(MetricDto metric, LiveMeasureDto measure) {
+    String textValue = measure.getTextValue();
+    Level qualityGate = Arrays.stream(Level.values()).filter(level -> level.name().equalsIgnoreCase(textValue)).findFirst()
+      .orElseThrow(() -> new IllegalStateException(format("Unknown quality gate status : '%s'", textValue)));
+    return svgGenerator.generateBadge(METRIC_NAME_BY_KEY.get(metric.getKey()), QUALITY_GATE_MESSAGE_BY_STATUS.get(qualityGate), COLOR_BY_QUALITY_GATE_STATUS.get(qualityGate));
+  }
+
+  private String generateRating(MetricDto metric, LiveMeasureDto measure) {
+    Double value = getNonNullValue(metric, measure, LiveMeasureDto::getValue);
+    Rating rating = valueOf(value.intValue());
+    return svgGenerator.generateBadge(METRIC_NAME_BY_KEY.get(metric.getKey()), rating.name(), COLOR_BY_RATING.get(rating));
+  }
+
+  private LiveMeasureDto getMeasure(DbSession dbSession, ComponentDto project, String metricKey) {
+    return dbClient.liveMeasureDao().selectMeasure(dbSession, project.uuid(), metricKey)
+      .orElseThrow(() -> new StickerException(format("Measure '%s' has not been found for project '%s' and branch '%s'", metricKey, project.getKey(), project.getBranch())));
+  }
+
+  private static <PARAM> PARAM getNonNullValue(MetricDto metric, LiveMeasureDto measure, Function<LiveMeasureDto, PARAM> function) {
+    PARAM value = function.apply(measure);
+    checkState(value != null, "Invalid measure on metric '%s'", metric.getKey());
+    return value;
   }
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. ([squid:S00119](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00119))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10263%2Fmeasure_badge&issues=AWD-l1Baf6JiG_HArjU7&open=AWD-l1Baf6JiG_HArjU7)

[ISSUE_KEY]: AWD-l1Baf6JiG_HArjU7

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED